src/plugins/data/public/search/search_interceptor/search_interceptor.ts

```diff
@@ -518,25 +517,17 @@ export class SearchInterceptor {
     if (e instanceof AbortError || e instanceof SearchTimeoutError) {
       // The SearchTimeoutError is shown by the interceptor in getSearchError (regardless of how the app chooses to handle errors)
       return;
-    } else if (e instanceof EsError) {
-      this.deps.toasts.addDanger({
-        title: i18n.translate('data.search.esErrorTitle', {
-          defaultMessage: 'Cannot retrieve search results',
-        }),
-        text: toMountPoint(e.getErrorMessage(this.application), { theme$: this.deps.theme.theme$ }),
-      });
-    } else if (e.constructor.name === 'HttpFetchError' || e instanceof BfetchRequestError) {
-      const defaultMsg = i18n.translate('data.errors.fetchError', {
-        defaultMessage: 'Check your network connection and try again.',
-      });
+    }
 
+    const overrideDisplay = getSearchErrorOverrideDisplay({
+      error: e,
+      application: this.application,
+    });
+
+    if (overrideDisplay) {
       this.deps.toasts.addDanger({
-        title: i18n.translate('data.search.httpErrorTitle', {
-          defaultMessage: 'Unable to connect to the Kibana server',
-        }),
-        text: toMountPoint(e.message || defaultMsg, {
-          theme$: this.deps.theme.theme$,
-        }),
+        title: overrideDisplay.title,
```

Comment:
It's unrelated to this PR, but just a thought, it would be interesting if when this is called (which is no longer the case in Discover 🥳 )

E.g. here:
https://github.com/elastic/kibana/blob/cb0a8057c8feb40e73695ad5748b7a882b6e2c77/src/plugins/visualizations/public/visualize_app/utils/get_visualization_instance.ts#L72

Reply:
I imagine calling `data.search.showError` too much probably is responsible for some of the toast storms. It might be a candidate to consider throttling too.